extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -339,27 +335,26 @@ export class UserNotificationManager {
    * Retrieve a promise associated to an `amp-user-notification` component
    * that is resolved when user agrees to the terms.
    * @param {string} id
-   * @return {!Promise}
+   * @return {!Promise<!AmpUserNotification>}
    */
   get(id) {
     this.managerReadyPromise_.then(() => {
       if (this.win_.document.getElementById(id) == null) {
         user.warn(TAG, `Did not find amp-user-notification element ${id}.`);
       }
     });
-    return this.getElementDeferById_(id).promise;
```

Comment:
looking at the code again, it shouldn't be a bug. although yes a bit weird. I don't think we need to be dependent on managerReady (viewer + dom ready) to return the promise (since that promise is resolved independently). this was really just to give the publisher dev a warning if they mistyped the notification "id". although you could argue as that being a user error instead of a warning and that we could move this dom scan to the url replacement code.